web/src/utilities/utilities.js

```diff
@@ -611,6 +611,38 @@ export const Utilities = {
     }
   },
 
+  isInitialAppInstall(app) {
+    if (!app) {
+      return false;
+    }
+
+    if (app.downstream?.currentVersion) {
+      // a version has already been deployed
+      return false;
+    }
+
+    if (
+      !app.downstream?.pendingVersions ||
+      app.downstream.pendingVersions.length === 0
+    ) {
+      // there's no version that's been deployed or is pending, so it's an initial install
+      return true;
+    }
+
+    const firstPendingVersion = app.downstream.pendingVersions[0];
+    if (
+      firstPendingVersion.status === "pending_cluster_management" ||
```

Comment:
would it make more sense to check the `pending`, `deploying`, `deployed`, and `failed` statuses instead, and return true if the status doesn't match either of them? just in case another status gets added in the future.

Reply:
e.g.: check if the version is not "deployable" yet

Reply:
actually, there should be a `isDeployable` field on the version that you can check. that might be sufficient here.

Reply:
this: https://github.com/replicatedhq/kots/blob/2c8ddb48f3e5f383e6658ee2e7aa373f3c07d4cf/pkg/api/downstream/types/types.go#L44

Reply:
actually, that field might not be set during that workflow to save DB queries. reference: https://github.com/replicatedhq/kots/blob/2c8ddb48f3e5f383e6658ee2e7aa373f3c07d4cf/pkg/api/downstream/types/types.go#L42-L43

Reply:
actually, that field might not be set during that workflow to save DB queries. reference: https://github.com/replicatedhq/kots/blob/2c8ddb48f3e5f383e6658ee2e7aa373f3c07d4cf/pkg/api/downstream/types/types.go#L42-L43

Reply:
good question... thinking out loud, i suppose that could have the same issue if we add a status that is _not_ part of the initial install flow since we would have to potentially add that one to the check. i'm ok with checking the statuses as they are now since worst case, we end up showing the navbar when we shouldn't instead of hiding it when it should be showing. lmk if i'm missing anything there and we need to change it